src/Generation/TestNameValueProducer.php

```diff
@@ -137,13 +137,36 @@ public function __construct($prod, $method, $f)
 
     // TODO: Refactor this to share code with very similar code in GapicClientExamplesGenerator.php
     // TODO: Handle nesting - see PublishSeries.
-    public function fieldInit(MethodDetails $method, FieldDetails $field, $fieldVar, string $fieldVarName, ?Variable $clientVar, Vector &$astAcc)
+    public function fieldInit(MethodDetails $method, FieldDetails $field, &$fieldVar, string $fieldVarName, ?Variable $clientVar, Vector &$astAcc)
     {
         if (!$field->isRequired) {
             $astAcc = $astAcc->append(null);
             return;
         }
 
+        if ($field->isOneOf) {
+            $oneofWrapperType = $field->toOneofWrapperType($method->serviceDetails->namespace);
+            // Initialize the oneof, e.g.
+            //   $supplementaryData = new SupplementaryDataOneof();
+            $oneofDescProto = $field->containingMessage->getOneofDecl()[$field->oneOfIndex];
+            $fieldVar = AST::var(Helpers::toCamelCase($oneofDescProto->getName()));
+            // TODO(v2): Enable recursion like the map source below. We don't do this yet
+            // because nobody exercises this use case.
+            $astAcc = $astAcc->append(AST::assign($fieldVar, AST::new($this->ctx->type($oneofWrapperType))()));
+            // Set the oneof field, e.g.
+            //    $supplementaryData->setExtraDescription('extraDescription-1352933811');
+            $astAcc = $astAcc->append(
+                AST::call(
+                    $fieldVar,
+                    AST::method("set" . Helpers::toUpperCamelCase($field->camelName))
+                )(
+                        // TODO(v2): Handle non-primitive types.
```

Comment:
The formatting seems a bit off on these lines.

Reply:
Done.